### ISSUE TITLE
[Bug] Ember Data Model's errors field is not autotracked in 3.20

### ISSUE THREAD

**Issue by andreyfel** [emberjs/ember.js#19097]
I'm updating Ember from 3.19 to 3.20 in my app and there is an issue which appeared after the update. I have a component which renders server errors for a certain field in the Ember Data model. In the component js it looks like this:

```
get errors() {
  const { model, field } = this.args;
  return model.errors.errorsFor(field).map(error => error.message);
}
```

Then I just render these messages in the template:
```
{{#each this.errors as |error|}}
  <li>{{error}}</li>
{{/each}}
```
If error comes after first render the component automatically re-rendered to show new errors.

It worked in Ember 3.19 but it doesn't work in Ember 3.20.

If I change it to computed by adding the following decorator it starts working in 3.20: `@computed('args.model.errors.[]')`

**Comment by andreyfel** [comment]
Note that it is `ember-source` version that makes the difference. If I use `ember-source@3.19` + `ember-data@3.20` it works.

**Comment by andreyfel** [comment]
Here is the link to the reproduction repo: https://github.com/andreyfel/ember-data-errors-autotracking-bug

**Comment by andreyfel** [comment]
I've tested the fix https://github.com/emberjs/data/pull/7273 in my app and it fixed the issue with appearing of the new error. Unfortunately, removing the error is not working :(
I've updated my reproduction repo https://github.com/andreyfel/ember-data-errors-autotracking-bug  to illustrate the issue.
@pzuraq do you have any idea?

**Comment by snewcomer** [comment]
@andreyfel I believe this issue can be closed now with `3.20.4` (or recently release `3.22.0`)

**Comment by andreyfel** [comment]
@snewcomer Yes! Thanks!
